src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -978,16 +977,215 @@ void CmdSketcherViewSection::activated(int iMsg)
+    auto* sketchView = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+    assert(sketchView);
+
+    if (iMsg == 0) {
+        bool newState = !a[0]->isChecked();
+        a[0]->setChecked(newState);
+        sketchView->ShowGrid.setValue(newState);
+
+        //activate/deactivate others
+        a[1]->setEnabled(newState);
+        a[2]->setEnabled(newState);
+    }
+    else if (iMsg == 1) {
+        bool newState = !a[1]->isChecked();
+        a[0]->setChecked(newState);
+        sketchView->GridSnap.setValue(newState);
+
+        pcAction->setProperty("defaultAction", QVariant(0));
+    }
+    else if (iMsg == 2){
+        bool newState = !a[2]->isChecked();
+        a[2]->setChecked(newState);
+        sketchView->GridAuto.setValue(newState);
+
+        a[3]->setEnabled(!newState);
+        if (!newState)
+            sketchView->GridSize.setValue(sketchView->GridSize.getValue()); //redraw grid.
+
+        pcAction->setProperty("defaultAction", QVariant(0));
+    }
+    else if (iMsg == 3) {
+        //nothing but don't return either.
+    }
+    else
+        return;
+
+    pcAction->setIcon(Gui::BitmapFactory().iconFromTheme(a[0]->isChecked() ? "Sketcher_GridToggle" : "Sketcher_GridToggle_Deactivated"));
+}
+
+Gui::Action* CmdSketcherCompGrid::createAction()
+{
+    Gui::ActionGroup* pcAction = new Gui::ActionGroup(this, Gui::getMainWindow());
+    pcAction->setDropDownMenu(true);
+    pcAction->setExclusive(false);
+    applyCommandData(this->className(), pcAction);
+
+    QAction* a0 = pcAction->addAction(QString());
+    QAction* a1 = pcAction->addAction(QString());
+    QAction* a2 = pcAction->addAction(QString());
+
+    GridSpaceAction* a3 = new GridSpaceAction(pcAction);
+    pcAction->addAction(a3);
+
+    a0->setCheckable(true);
+    a1->setCheckable(true);
+    a2->setCheckable(true);
+
+    _pcAction = pcAction;
+    languageChange();
+
+    return pcAction;
+}
+
+void CmdSketcherCompGrid::languageChange()
+{
+    Command::languageChange();
+
+    if (!_pcAction)
+        return;
+    Gui::ActionGroup* pcAction = qobject_cast<Gui::ActionGroup*>(_pcAction);
+    QList<QAction*> a = pcAction->actions();
+
+    QAction* a0 = a[0];
+    a0->setText(QApplication::translate("CmdSketcherGridToggle", "Activate Grid"));
+    a0->setToolTip(QApplication::translate("CmdSketcherGridToggle", "Activate Grid"));
+    a0->setStatusTip(a0->toolTip());
+    QAction* a1 = a[1];
+    a1->setText(QApplication::translate("CmdSketcherGridSnap", "Grid Snap"));
+    a1->setToolTip(QApplication::translate("CmdSketcherGridSnap", "New points will snap to the nearest grid line.\nPoints must be set closer than a fifth of the grid size to a grid line to snap."));
+    a1->setStatusTip(a1->toolTip());
+    QAction* a2 = a[2];
+    a2->setText(QApplication::translate("CmdSketcherGridAuto", "Grid Auto Resize"));
+    a2->setToolTip(QApplication::translate("CmdSketcherGridAuto", "Resize grid automatically depending on zoom."));

```

Comment:
Would be better to use only one context for all translation, something like "CmdSketcherCompGrid".

Reply:
Oh yes, the other commands have been deleted so all need to use this context indeed.